inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractStreamSourceOperation.java

```diff
@@ -126,11 +126,6 @@ public void updateOpt(SourceRequest request, Integer groupStatus, String operato
         // Setting updated parameters of stream source entity.
         setTargetEntity(request, entity);
         entity.setVersion(entity.getVersion() + 1);
-        if (GroupState.forCode(groupStatus).equals(GroupState.CONFIG_SUCCESSFUL)) {
```

Comment:
Why remove those lines?

Reply:
If the source was normal or failed, after the update operation, the status should be transfer to `new` or `to_be_add`.

Reply:
if the source was frozen, then you change to new or to_be_add, the status is fuzed

Reply:
change like this
 // If failed , return to new so can be init again
        if (SourceState.forCode(entity.getStatus()) == SourceState.SOURCE_FAILED){
            entity.setStatus(SourceState.SOURCE_NEW.getCode());
        }

Reply:
Should we need concerned about the group status?
For example, if the Group was SUCCESS or FAILED, how to control the Source status?

Reply:
We can handle it like this: do not modify the state when modifying the source info, and then call the restart interface after the modification.

Reply:
done